app/assets/src/components/views/report/ReportTable/DetailCells.jsx

```diff
@@ -4,67 +4,79 @@ import { get } from "lodash/fp";
 // TODO(mark): Use alias instead, to avoid dots.
 import PropTypes from "../../../utils/propTypes";
 
-const DetailCells = ({
-  taxons,
-  taxonRowRefs,
-  renderName,
-  renderNumber,
-  showConcordance,
-  getRowClass,
-  onTaxonClick,
-  reportDetails,
-  backgroundData,
-  showAssemblyColumns
-}) => {
-  return taxons.map(taxInfo => (
-    <tr
-      key={taxInfo.tax_id}
-      id={`taxon-${taxInfo.tax_id}`}
-      ref={elm => {
-        taxonRowRefs[taxInfo.tax_id] = elm;
-      }}
-      className={getRowClass(taxInfo)}
-    >
-      <td>
-        {renderName(taxInfo, reportDetails, backgroundData, onTaxonClick)}
-      </td>
-      {renderNumber(
-        taxInfo.NT.aggregatescore,
-        null,
-        0,
-        true,
-        undefined,
-        taxInfo.topScoring == 1,
-        "score-column"
-      )}
-      {renderNumber(taxInfo.NT.zscore, taxInfo.NR.zscore, 1)}
-      {renderNumber(taxInfo.NT.rpm, taxInfo.NR.rpm, 1)}
-      {renderNumber(taxInfo.NT.r, taxInfo.NR.r, 0)}
-      {showAssemblyColumns &&
-        renderNumber(
-          get("summaryContigCounts.NT.contigs", taxInfo) || 0,
-          get("summaryContigCounts.NR.contigs", taxInfo) || 0,
-          0
+export default class DetailCells extends React.Component {
+  render() {
+    const {
+      taxons,
+      taxonRowRefs,
+      renderName,
+      renderNumber,
+      showConcordance,
+      getRowClass,
+      onTaxonClick,
+      reportDetails,
+      backgroundData,
+      showAssemblyColumns,
+      handleMouseEnter,
+      handleMouseLeave
+    } = this.props;
+
+    return taxons.map(taxInfo => (
+      <tr
+        key={taxInfo.tax_id}
+        id={`taxon-${taxInfo.tax_id}`}
+        ref={elm => {
+          taxonRowRefs[taxInfo.tax_id] = elm;
+        }}
+        className={getRowClass(taxInfo)}
+        onMouseEnter={() => handleMouseEnter(taxInfo.tax_id)}
+        onMouseLeave={() => handleMouseLeave(taxInfo.tax_id)}
```

Comment:
Can `taxInfo` ever be `undefined`? We have special rows like "All taxa with neither family nor genus classification" and "Non-genus-specific reads in family Sinobacteraceae" which don't correspond to a single taxon. Also, can `taxInfo.tax_id` be `null` or `undefined` for multiple rows? (I don't know the answer, but it would be good to check.) If so, that might cause multiple rows to show their `<HoverActions>` when you hover on any single one of them.

Reply:
“Non-specific” entries correspond to negative taxids. I don’t think they should be undefined